src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -4,27 +4,30 @@ class Webui::Users::NotificationsController < Webui::WebuiController
 
   # TODO: Remove this when we'll refactor kerberos_auth
   before_action :kerberos_auth
-  before_action :check_param_type, :check_param_project, only: :index
-
+  before_action :set_current_user, only: %i[index update]
   after_action :verify_policy_scoped
 
   def index
+    @selected_filter = params.permit!.to_h.except(*%w[controller action])
     @notifications = paginated_notifications
     @show_read_all_button = show_read_all_button?
-    @filtered_project = Project.find_by(name: params[:project])
-    @selected_filter = selected_filter
-    @current_user = User.session
+    # This is a GET form, we're not going to update anything so it's safe to permit any params
+    @filtered_by = @selected_filter.filter { |_k, v| v.present? }.keys - %w[unread read]
   end
 
   def update
+    @selected_filter = params.permit!.to_h.except(*%w[notification_ids controller action])
     notifications = if params[:update_all]
                       fetch_notifications
                     else
                       fetch_notifications.where(id: params[:notification_ids])
                     end
     # rubocop:disable Rails/SkipsModelValidations
-    read_count = notifications.where(delivered: false).update_all('delivered = !delivered')
-    unread_count = notifications.where(delivered: true).update_all('delivered = !delivered')
+    # FIXME: This has room for improvement
```

Comment:
How do you expect to pay back this technical debt?

Reply:
We're going to attempt to reduce this on another PR

Reply:
Here https://trello.com/c/HklBDh6i/146-improve-the-code-that-marks-read-and-unread-notifications